src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -227,13 +237,50 @@ public void validate(ClientState state)
             Guardrails.tables.guard(totalUserTables + 1, targetTableName, false, state);
         }
         validateDefaultTimeToLive(attrs.asNewTableParams());
+
+        if (createLikeOptions.contains(ALL) && createLikeOptions.size() > 1)
+        {
+            throw ire("It is not allowed to use ALL option together with " + createLikeOptions.stream().filter(option -> option != ALL).sorted().collect(Collectors.toList()));
+        }
+    }
+
+    private void cloneIndexesAndTriggersIfNeeded(TableMetadata.Builder builder, TableMetadata sourceTableMeta, KeyspaceMetadata targetKeyspaceMeta, String targetTableName)
+    {
+        if (createLikeOptions.contains(INDEX) ||
+            createLikeOptions.contains(ALL))
+        {
+            Indexes.Builder idxBuilder = Indexes.builder();
+            for (IndexMetadata indexMetadata : sourceTableMeta.indexes)
+            {
+                // todo use source index name if target keyspace don't have same index name but use random name now as cep-43 described
+                String baseName = targetKeyspaceMeta.name + "_" + targetTableName + "_" + indexMetadata.name;
+                String idxName = targetKeyspaceMeta.findAvailableIndexName(baseName);
+                idxBuilder.add(IndexMetadata.fromSchemaMetadata(idxName, indexMetadata.kind, indexMetadata.options));
```

Comment:
I am a bit confused by the logic surrounding the index and trigger name. I fear that it will bring more confusion than anything else. If an index with the same name already exists we should probably raise an error and let the user deal with it, no?